component/newstorage/mem/go.mod

```diff
@@ -0,0 +1,17 @@
+// Copyright SecureKey Technologies Inc. All Rights Reserved.
```

Comment:
Made this a separate package for two reasons:
1. To avoid a cyclic dependency with the new common tests file
2. So projects that need to just use the storage from Aries can have a dependency on just the storage stuff